content/blog/writing-crossguard-policies/index.md

```diff
@@ -0,0 +1,167 @@
+2. `StackValidationPolicy` - Validates the stack as a whole.
+
+Let's take a closer look at these and why you'd use one over the other.
+
+## Resource Validation Policies
+
+Policies of `ResourceValidationPolicy` are called for each resource in a stack. These policies are run _before_ a resource is registered and thus block an out-of-compliance resource from ever being created or modified.
+
+A resource validation function is passed `args` with more information about the resource (such as the resource's type, _input_ properties, name, and URN) and a `reportViolation` callback that can be called to report a policy violation. In most cases, you will just use the helper function `validateTypedResource` to filter the resource type you want to validate and provide stronger typing of the resource's input properties.
+
+An example resource validation policy is shown below:
+
+```typescript
+const ec2DetailedMonitoring: ResourceValidationPolicy = {
+    name: "ec2-detailed-monitoring",
+    description: "Detailed monitoring should be enabled for all EC2 instances.",
+    enforcementLevel: "mandatory",
+    validateResource: validateTypedResource(aws.ec2.Instance, (instance, args, reportViolation) => {
+        if (instance.monitoring !== true) {
+            reportViolation("EC2 Instance should have monitoring enabled.");
+        }
+    }),
+};
+```
+
+If you have multiple resources that require a similar policy, you can group them together under one policy. This is possible by setting `validateResource` to an array of callback functions. The example below shows a case where we want to run similar checks against multiple types of resources.
+
+```typescript
+const elbLoggingEnabled: ResourceValidationPolicy = {
+    name: "elb-logging-enabled",
+    description: "Checks whether the Application Load Balancers and the Classic Load Balancers have logging enabled.",
+    enforcementLevel: "mandatory",
+    validateResource: [
+        validateTypedResource(aws.elasticloadbalancing.LoadBalancer, (loadBalancer, args, reportViolation) => {
+            if (loadBalancer.accessLogs === undefined || !loadBalancer.accessLogs.enabled) {
+                reportViolation("Elastic Load Balancer must have access logs enabled.");
+            }
+        }),
+        validateTypedResource(aws.elasticloadbalancingv2.LoadBalancer, (loadBalancer, args, reportViolation) => {
+            if (loadBalancer.accessLogs === undefined || !loadBalancer.accessLogs.enabled) {
+                reportViolation("Elastic Load Balancer must have access logs enabled.");
+            }
+        }),
+        validateTypedResource(aws.applicationloadbalancing.LoadBalancer, (loadBalancer, args, reportViolation) => {
+            if (loadBalancer.accessLogs === undefined || !loadBalancer.accessLogs.enabled) {
+                reportViolation("Application Load Balancer must have access logs enabled.");
+            }
+        }),
+    ],
+};
+```
+
```

Comment:
We should add a quick paragraph about the drawbacks of a resource validation policy too. For example, since it is ran before the resource is created, you cannot verify things like the assigned name or properties the "live" state of the resource has.

So for example, some types of policies cannot be expressed in a resource validation policy. But the good news is that there are other types of policies you can write...